asynq/decorators.pyi

```diff
@@ -31,18 +33,24 @@ def get_async_or_sync_fn(fn: object) -> Any: ...
 class PureAsyncDecoratorBinder(qcore.decorators.DecoratorBinder):
     def is_pure_async_fn(self) -> bool: ...
 
-class PureAsyncDecorator(qcore.decorators.DecoratorBase, Generic[_T]):
+class PureAsyncDecorator(
+    qcore.decorators.DecoratorBase, Generic[_T, OriginalFunctionParams]
+):
     binder_cls = PureAsyncDecoratorBinder
     def __init__(
         self,
-        fn: Callable[..., _T],
+        fn: Callable[OriginalFunctionParams, _T],
         task_cls: Optional[Type[futures.FutureBase]],
         kwargs: Mapping[str, Any] = ...,
         asyncio_fn: Optional[_CoroutineFn] = ...,
     ) -> None: ...
     def name(self) -> str: ...
     def is_pure_async_fn(self) -> bool: ...
-    def asyncio(self, *args, **kwargs) -> _Coroutine: ...
+    def asyncio(
+        self,
+        *args: OriginalFunctionParams.args,
+        **kwargs: OriginalFunctionParams.kwargs,
+    ) -> _Coroutine: ...
```

Comment:
This loses the return type, use `Coroutine[Any, Any, _T]`

Reply:
I see. This is perhaps fixable with overloads: if the decorator is passed something returning a `Generator[Any, Any, T]`, then this returns `T`, else it returns the return type directly.

Reply:
This seems important to fix because otherwise we lose type checking for the object returned from the asynq function.